pootle/apps/pootle_data/language_data.py

```diff
@@ -0,0 +1,30 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) Pootle contributors.
+#
+# This file is a part of the Pootle project. It is distributed under the GPL3
+# or later license. See the LICENSE file for a copy of the license and the
+# AUTHORS file for copyright and authorship information.
+
+from .models import TPData
+from .utils import RelatedStoresDataTool
+
+
+class LanguageDataTool(RelatedStoresDataTool):
+    """Retrieves aggregate stats for a Language"""
+
+    group_by = (
+        "tp__language__code",
+        "tp__project__code")
+
+    @property
+    def data_model(self):
+        return TPData.objects
+
+    @property
+    def stat_data(self):
+        return self.data_model.filter(
+            tp__language=self.context)
+
+    def get_root_child_path(self, child):
+        return "-".join(child[field] for field in self.group_by)
```

Comment:
k - we can revisit this - i changed it to this to try and make it clearer, but obv didnt work. Any suggestions?